lib/Serialization/SerializeSIL.cpp

```diff
@@ -222,7 +222,7 @@ namespace {
 
     /// Helper function to determine if given the current state of the
     /// deserialization if the function body for F should be deserialized.
-    bool shouldEmitFunctionBody(const SILFunction *F);
+    bool shouldEmitFunctionBody(const SILFunction *F, bool isReference = false);
```

Comment:
Nitpick: The only place where this is false is `addMandatorySILFunction`, which itself only actually makes sense when in "emit -Onone prespecializations" mode. Maybe this shouldn't have a default, or should default to true?